### PR TITLE
Add NHDD mappings for the Immuniztion service

### PR DESCRIPTION
**Pull request by freddieptf**
Part of #110, building on #114

@kitsao @derickl I couldn't find most of the mappings required for vaccines and vitamin A doses. This is mostly for vaccines that have follow ups like opv0, opv1, opv2 etc. Can we involve a designer or someone with more info on where to find these codes? I have used place holders here so the testing on https://chis-staging.health.go.ke can go on

### CODE REVIEW

File: src/utils/referral.js [hunk 1]
```diff
@@ -323,7 +324,7 @@ const generateFHIRServiceRequest = (dataRecord) => {
         },
       },
     ],
-    reasonCode: extractReasonCode(dataRecord.screening),
+    reasonCode: extractReasonCode(dataRecord.screening, dataRecord.service),
     note: extractNotes(dataRecord.supportingInfo),
   };
   return FHITServiceRequest;
```
Comment (derickl) [inline src/utils/referral.js:330]:
```suggestion
  return FHIRServiceRequest;
```

File: src/utils/referral.js [hunk 2]
```diff
@@ -255,7 +256,7 @@ const extractReasonCode = (data) => {
     ),
   ].filter((elem) => elem !== `none`);
   reasons.forEach((reason) => {
-    const coding = echisNHDDValuesCoding[reason];
+    const coding = !!service ? serviceMapping[service].mapping[reason] : echisNHDDValuesCoding[reason];
```
Comment (derickl) [inline src/utils/referral.js:259]: Could you use Typescripts syntax

e.g `serviceMapping[service]?.mapping[reason]` ?

File: src/controllers/referral.js [hunk 3]
```diff
@@ -168,7 +168,7 @@ const createTaskReferral = async (serviceRequest) => {
     if (data.rows.length > 0) {
       logger.information(CLIENT_FOUND_REPORT_IN_ECHIS);
       const patientDoc = data.rows[0].value;
-      const notesDeserialize = JSON.parse(serviceRequest?.note[0].text); //Remove backslash and parse JSON
+      const notesDeserialize = serviceRequest?.note[0].text;
```
[the remark this answers is not in the document]
Comment (freddieptf) [reply src/controllers/referral.js:171]: I'd assume all requests of that type have a note with follow up instructions. I had to remove the parsing as it was trying to parse a JS object and throwing errors which killed the app. I'm not sure this should be in the codebase right now, might have been used previously for testing

**Review by derickl — APPROVED** [review]
Looks good

**Pull request merged**